api/pkg/template/config.go

```diff
@@ -0,0 +1,260 @@
+	if err != nil {
+		return "", fmt.Errorf("decode base64 value: %w", err)
+	}
+	return string(decoded), nil
+}
+
+func (e *Engine) configOptionEquals(name, expected string) (bool, error) {
+	e.recordDependency(name)
+
+	resolved, err := e.resolveConfigItem(name, nil)
+	if err != nil {
+		return false, fmt.Errorf("resolve config item: %w", err)
+	}
+	return resolved.Effective == expected, nil
+}
+
+func (e *Engine) configOptionNotEquals(name, expected string) (bool, error) {
+	e.recordDependency(name)
+
+	resolved, err := e.resolveConfigItem(name, nil)
+	if err != nil {
+		// NOTE: this is parity from KOTS but I would expect this to return true
+		return false, fmt.Errorf("resolve config item: %w", err)
+	}
+	return resolved.Effective != expected, nil
+}
+
+func (e *Engine) configOptionFilename(name string) (string, error) {
+	e.recordDependency(name)
+
+	resolved, err := e.resolveConfigItem(name, e.getConfigItemFilename)
+	if err != nil {
+		return "", fmt.Errorf("resolve config item: %w", err)
+	}
+	return resolved.Effective, nil
+}
+
+// resolveConfigItem processes a config item and returns its resolved values. It determines:
+// 1. The effective value - the final value used in templates
+// 2. The templated value - the templated result of the item's "value" field
+// 3. The templated default - the templated result of the item's "default" field
+//
+// The effective value is determined either by:
+// - The provided effectiveValueGetter function, if specified
+// - Or following priority: user-provided value > config value > default value
+func (e *Engine) resolveConfigItem(name string, effectiveValueGetter func(configItem *kotsv1beta1.ConfigItem) (string, error)) (*ResolvedConfigItem, error) {
+	// Check if we have a cached value
+	if cacheVal, ok := e.getItemCacheValue(name); ok {
+		return cacheVal, nil
+	}
+
+	// Check for circular dependency
+	if slices.Contains(e.stack, name) {
+		return nil, fmt.Errorf("circular dependency detected for %s", name)
+	}
+
+	// Track resolution path for dependency discovery
+	e.stack = append(e.stack, name)
+	defer func() {
+		if len(e.stack) > 0 {
+			e.stack = e.stack[:len(e.stack)-1]
+		}
+	}()
+
+	// Find the config item definition
+	configItem := e.findConfigItem(name)
+	if configItem == nil {
+		return nil, fmt.Errorf("config item %s not found", name)
+	}
+
+	var effectiveValue, templatedValue, templatedDefault string
+
+	// Template the value field if present
+	if configItem.Value.String() != "" {
+		val, err := e.processTemplate(configItem.Value.String())
+		if err != nil {
+			return nil, fmt.Errorf("template value for %s: %w", name, err)
+		}
+		templatedValue = val
+	}
+
+	// Template the default field if present
+	if configItem.Default.String() != "" {
+		val, err := e.processTemplate(configItem.Default.String())
+		if err != nil {
+			return nil, fmt.Errorf("template default for %s: %w", name, err)
+		}
+		templatedDefault = val
+	}
```

Comment:
This seems like it doesn't need to run if effectiveValueGetter is not nil

Reply:
Why? The value and default are used when templating individual config items that don't need the effective value.